openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -417,6 +418,95 @@ def schedules_for_bin(self):
                 yield (user, schedule.enrollment.course.closest_released_language, template_context, course.self_paced)
 
 
+@attr.s
+class CourseNextSectionUpdate(PrefixedDebugLoggerMixin, RecipientResolver):
+    """
+    Send a message to all users whose schedule gives them a due date of yesterday.
+    """
+    async_send_task = attr.ib()
+    site = attr.ib()
+    target_datetime = attr.ib()
+    course_key = attr.ib()
+    override_recipient_email = attr.ib(default=None)
+
+    log_prefix = 'Next Section Course Update'
+    experience_filter = Q(experience__experience_type=ScheduleExperience.EXPERIENCES.course_updates)
+
+    def send(self):
+        schedules = self.get_schedules()
+        LOG.info(
+            u'Found {} emails to send for course-key: {}'.format(
+                len(schedules), self.course_key
+            )
+        )
+        for (user, language, context, is_self_paced) in schedules:
+            msg_type = CourseUpdate() if is_self_paced else InstructorLedCourseUpdate()
+            msg_type.personalize(
+                Recipient(
+                    user.username,
+                    self.override_recipient_email or user.email,
+                ),
+                language,
+                context,
+            )
+            LOG.info(
+                u'Sending email to user: {} for course-key: {}'.format(
+                    user.username,
+                    self.course_key
+                )
+            )
+            # TODO: Uncomment below when going live
+            # with function_trace('enqueue_send_task'):
+            #     self.async_send_task.apply_async((self.site.id, str(msg)), retry=False)
+
+    def get_schedules(self):
+        target_date = self.target_datetime.date()
+        schedules = get_schedules_with_due_date(self.course_key, target_date).filter(
+            self.experience_filter,
+            active=True,
+            enrollment__user__is_active=True,
+        )
+
+        template_context = get_base_template_context(self.site)
+        for schedule in schedules:
+            enrollment = schedule.enrollment
+            course = schedule.enrollment.course
+            user = enrollment.user
+
+            try:
+                week_highlights, week_num = get_next_section_highlights(user, course.id, target_date)
+            except CourseUpdateDoesNotExist:
+                LOG.warning(
+                    u'Weekly highlights for user {} of course {} does not exist or is disabled'.format(
+                        user, course.id
+                    )
+                )
+                # continue to the next schedule, don't yield an email for this one
+            else:
```

Comment:
Nit: I think it's just personal preference, but I'd rather remove the `else` and either put it into the try or just dedent the logic and add in a `continue` into the except block. Easier for me to read that way, but definitely a nit.